sleap/io/cameras.py

```diff
@@ -853,13 +864,62 @@ def update_points(
             exclude_complete: If True, then do not update points that are marked as
                 complete. Default is True.
         """
+        # Ensure we are working with a float array
+        points_3d = points_3d.astype(float)
+
+        # Check if points are 3D
+        is_3d = points_3d.shape[-1] == 3
+        if not is_3d:
+            raise ValueError("Expected 3D points with shape (M, T, N, 3).")
+        
+        # Check that the correct shape was passed in
+        n_views, n_instances, n_nodes, n_coords = points_3d.shape
+        assert n_views == len(
+            self.cams_to_include
+        ), f"Expected {len(self.cams_to_include)} views, got {n_views}."
+        assert n_instances == len(
+            instance_groups
+        ), f"Expected {len(instance_groups)} instances, got {n_instances}."
+        assert n_coords == 3, f"Expected 3 coordinates, got {n_coords}."
+
+        # Reproject 3D points into 2D points for each camera view
+        pts_reprojected = reproject(
+            points_3d,
+            calib=self.session.camera_cluster,
+            excluded_views=self.excluded_views,
+        )  # M=include x F=1 x T x N x 2
+
+        # Squeeze back to the original shape
+        points_reprojected = np.squeeze(pts_reprojected, axis=1)  # M=include x TxNx2
+
+        # Get projection bounds (based on video height/width)
+        bounds = self.session.projection_bounds
```

Comment:
Let's also pass in an input argument `projection_bounds` since there is no `InstanceGroup.session` attribute.